### DISCUSSION TITLE
Fallback glyphs and sprite URL

### DISCUSSION THREAD

**Discussion by ttmx** [maplibre/maplibre-gl-js#3514] [Q&A]
Hi,
we are building an application where each of the tileservers don't have perfect availability, but we have multiple tileservers, which together provide nice availability.

It is simple to add an extra tile URL, but I do not know how to add more urls for glyphs and sprites in case the main server is not up.
How can one do this? I could not find any information about this.

Thank you for the great lib!

**Comment by HarelM** [comment]
I would recommend a backend proxy to allow high availability.
But in theory you can solve this using `addProtocol` in order to try and fetch from all servers and return the one that returned a valid answer.
Note that addProtocol implementation in 4.x which is about to be released soon has changed a bit.

**Reply by ttmx** [reply]
The issue is that we have no one server we can rely on. We're running 3 instances of tileserver-gl in the home of 3 different people. Each one of us has 98% uptimes, if we could just fall back to a second person's server, we could have near 100% :)
I understand that the list of tileservers does not do what I thought it did, I tested it after I posted and realized that.
How would you do this? We can have manual checks to see if the servers are up, and use that url, but that's adds one RTT to loading times, which we would rather avoid.

I understand this is a very particular case.
Thank you!

**Comment by neodescis** [comment]
The ability to add additional URLs is not to provide fallback servers, but rather to allow the browser to make more (tile) requests by bypassing the browser per-host request limit. It's basically rudimentary client-side load balancing. So, you would need to do something like @HarelM suggests to get true redundancy.
